_posts/2020-12-24-how-to-start-with-foreman

```diff
@@ -0,0 +1,399 @@
+
+#### A brief note on managed hosts
+
+When you import the Host as managed a lot more information is required to be filled in manually.
+For some fields you may not even have objects defined in Foreman yet. The bare minimum that you'll need to set is
+
+* architecture
+* operating system
+* installation medium
+* partition table
+* root password
+* domain
+* subnet
+
+*Architecture* can be customized, but usually, one of the default records will do. In the default installation
+usually one *Operating System* already exists. It's the OS of the host running Foreman.
+It's common to create more OSes under *Host -> Operating systems*. For that, you will need to select
+some partition table and installation medium. All information is required for provisioning, but we won't go
+to that details in this post.
+
+Root password is also meant to be used during provisioning. If you don't want to specify
+it for each host, set the default value in *Administer -> Settings -> Provisioning ->
+Root password*.
+
+*Domain* is automatically created if the machine name contains it. *Subnet* is required,
+if you're going to fill in the IP address of the host. Make sure you define *Subnet* under
+*Infrastructure -> Subnets* with the right IP range first.
+
+Now that has been a lot of clicking, but from now on, you are able to create managed
+hosts.
+
+If you already created hosts manually as unmanaged, you can use another
+compute resource function - Associate VMs. Not all compute resources support it and
+each uses a different mechanism, but usually, hosts are matched with compute resource
+VMs by MAC address. If the match is found, the *Host* is linked with VM.
+
+**Beware**, once you link host with VM, when you delete the host, it also deletes
+the VM on compute resource. If you want to keep VM untouched, either first disassociate
+it from host edit form or make sure to keep the global setting under *Administer -> Settings ->
+Provisioning -> Destroy associated VM on host delete* set to *No*.
+
+### The fruit
+
+If you follow one of the way to populate your inventory, now it's the time to see,
+how to perform some basic operations mentioned in the beginning.
+
+*how many CentOS 7 and 8 are running in your datacenter?*
+
+Go to Hosts overview page *Hosts -> All hosts*, type the following into the search field
+`os_title ~ CentOS and (os_major = 8 or os_major = 7)`.
+The table will filter only matching hosts, the total count is displayed at the bottom of the table.
+You can investigate the search field capabilities through its autocompletion feature.
+
+*search for all hosts in subnet 192.168.0.1/24*
+
+On the same page, you can try several ways to find hosts matching this subnet. If you
+defined the *Subnet* object and gave it a name, e.g. "lan1", you can search all hosts assigned
+to it via their primary interface like this
+
+`subnet.name = lan1`
+
+The other option is to search by IP adresses using a wildcard, e.g.
```

Comment:
We don't support searching by CIDR range, right? So /24 is the only one that would really work (or /16 or /8, but those are less likely to be used).

Reply:
yes, AFAIK it's not the limitation of how we store it, but rather scoped search lacking support for IPAddr type